CriteoPublisherSdk/Tests/IntegrationTests/CRInterstitialTests.m

```diff
@@ -59,10 +61,12 @@ - (void)setUp {
 
   self.urlOpener = CR_URLOpenerMock.new;
 
-  CR_DependencyProvider *dependencyProvider = CR_DependencyProvider.testing_dependencyProvider;
-
   self.displaySizeInjector = OCMClassMock([CR_DisplaySizeInjector class]);
+  self.integrationRegistry = OCMClassMock([CR_IntegrationRegistry class]);
```

Comment:
FAR: Would make sense to do this mocking by default in DependencyProvider+Testing

Reply:
The issue is that some tests would want to have mocks while other would want real instances or even spies.
On Android we make this easier with annotations and injection via a junit rule (https://github.com/criteo/android-publisher-sdk/blob/main/publisher-sdk-tests/src/androidTest/java/com/criteo/publisher/integration/IntegrationRegistryTest.kt#L52 for instance).

Maybe we could find a way to make this nicer in obj-c too ?

Reply:
In that specific case, the registry should be tested directly with a unit test, when it comes to integration tests it does not make sense and should always be mocked, thus the default init, mock that could be overrided in a specific test anyway. The Java annotations are syntactic sugar for initializations, but the concept is the same, isn't?
Anyway FAR

Reply:
Ok, but that may be good to have a solution that fit both integration and unit tests :D

About the Java annotations, no it's just annotation acting as a marker. Then there is an annotation processor looking for annotated elements and injecting mock/spy/instance in fields and the dependency provider: https://github.com/criteo/android-publisher-sdk/blob/main/test-utils/src/main/java/com/criteo/publisher/mock/DependenciesAnnotationInjection.java